libraries/AP_ExternalAHRS/AP_ExternalAHRS_MicroStrain7.cpp

```diff
@@ -191,29 +191,26 @@ void AP_ExternalAHRS_MicroStrain7::post_filter() const
     }
 
     for (int instance = 0; instance < NUM_GNSS_INSTANCES; instance++) {
-        // *INDENT-OFF*
-        AP_ExternalAHRS::gps_data_message_t gps {
-            gps_week: filter_data.week,
-            ms_tow: filter_data.tow_ms,
-            fix_type: (uint8_t) gnss_data[instance].fix_type,
-            satellites_in_view: gnss_data[instance].satellites,
-
-            horizontal_pos_accuracy: gnss_data[instance].horizontal_position_accuracy,
-            vertical_pos_accuracy: gnss_data[instance].vertical_position_accuracy,
-            horizontal_vel_accuracy: gnss_data[instance].speed_accuracy,
-
-            hdop: gnss_data[instance].hdop,
-            vdop: gnss_data[instance].vdop,
-
-            longitude: filter_data.lon,
-            latitude: filter_data.lat,
-            msl_altitude: gnss_data[instance].msl_altitude,
-
-            ned_vel_north: filter_data.ned_velocity_north,
-            ned_vel_east: filter_data.ned_velocity_east,
-            ned_vel_down: filter_data.ned_velocity_down,
-        };
-        // *INDENT-ON*
+        AP_ExternalAHRS::gps_data_message_t gps;
```

Comment:
Unless something else is zoering this you may end up using uninitialised stack data
```suggestion
        AP_ExternalAHRS::gps_data_message_t gps {};
```

... why are you changing from the current initialization structure?  What's the flash cost of that?

Reply:
OBE